### PR TITLE
add the frozen library and get rid of some hacky code for the actionM…

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add the frozen library to HELICS 3 and start to use it to replace a few hacks that were in place before.  

### Proposed changes

- add frozen lib
- use in ActionMessage
- remove a deprecated function from the Broker.hpp interface

### CODE REVIEW

File: interfaces/python/CMakeLists.txt.bak [hunk 1]
```diff
@@ -0,0 +1,268 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (nightlark) [inline interfaces/python/CMakeLists.txt.bak:1]: This file looks like it was accidentally added?

File: src/helics/core/ActionMessage.cpp [hunk 2]
```diff
@@ -573,13 +582,15 @@ std::unique_ptr<Message> createMessageFromCommand(ActionMessage&& cmd)
     return msg;
 }
 
+
 static constexpr char unknownStr[] = "unknown";
 
 // done in this screwy way because this can be called after things have started to be deconstructed
 // so static consts can cause seg faults someday will change to frozen::  once we can use all of
 // C++14
```
Comment (nightlark) [inline src/helics/core/ActionMessage.cpp:591]: Does this PR resolve this comment?

File: interfaces/python/CMakeLists.txt.bak (hunk 1 above)
Comment (phlptp) [reply interfaces/python/CMakeLists.txt.bak:1]: I will remove it.

File: src/helics/core/ActionMessage.cpp (hunk 2 above)
Comment (phlptp) [reply src/helics/core/ActionMessage.cpp:591]: yes I will remove it

File: src/helics/core/ActionMessage.cpp [hunk 3]
```diff
@@ -579,152 +587,131 @@ static constexpr char unknownStr[] = "unknown";
...
```
Comment (nightlark) [inline src/helics/core/ActionMessage.cpp:588]: The comment re: switching to frozen is still here.

**Pull request merged**